### ISSUE TITLE
Y-turns in Turn mode for headlands

### ISSUE THREAD

**Issue by Messy71** [Courseplay/courseplay#3088]
### Courseplay v6°01°00026
In turn mode for headlands tractor doesn't consider Turning circle. 
I've tried with :

- (auto) 4m;

- (mnl) 20m;

- (mnl) 60m;

- and for some odd reason (mnl) 13337m.
Here are some images and log's with debugging channels 12, 13, 14 :[F07Sowing.rar](https://drive.google.com/open?id=1gBSxZKCJpAM77_eJOYBJ3toGHGxlk-H9)

**PS :**
_I think the problem reveals itself when steering in reverse_

**Comment by pvajko** [comment]
That's a 500MB file compressed, I won't download it. Post images here, the log to gist, see https://github.com/Courseplay/courseplay/wiki

**Comment by Messy71** [comment]
[TurnHeadland(auto)4m-log.txt](https://gist.github.com/bff788664c86c59e2edb5e46b8663732.git)
![fsscreen_2019_02_02_18_02_53](https://user-images.githubusercontent.com/47163914/52171606-08d48580-2760-11e9-9987-dc303b0de870.png)
![fsscreen_2019_02_02_18_03_02](https://user-images.githubusercontent.com/47163914/52171607-08d48580-2760-11e9-8df7-cd663a2d661d.png)
![fsscreen_2019_02_02_18_03_14](https://user-images.githubusercontent.com/47163914/52171608-08d48580-2760-11e9-98a1-d7d7508d5aaf.png)
![fsscreen_2019_02_02_18_03_18](https://user-images.githubusercontent.com/47163914/52171610-08d48580-2760-11e9-9dae-6ff8834c145c.png)
![fsscreen_2019_02_02_18_03_26](https://user-images.githubusercontent.com/47163914/52171611-096d1c00-2760-11e9-96f0-31b3fcd547ac.png)
![fsscreen_2019_02_02_18_06_38](https://user-images.githubusercontent.com/47163914/52171612-096d1c00-2760-11e9-9488-b687c7572738.png)
![fsscreen_2019_02_02_18_07_04](https://user-images.githubusercontent.com/47163914/52171613-096d1c00-2760-11e9-8ee6-de62512f8350.png)
![fsscreen_2019_02_02_18_07_36](https://user-images.githubusercontent.com/47163914/52171614-096d1c00-2760-11e9-922b-b2363ceecdd5.png)

**Comment by Messy71** [comment]
[F07 SecondTry-Turn Headland- 20m.txt](https://gist.github.com/25097c5018b5d407d2088c19f87764a8.git)
![fsscreen_2019_02_02_18_29_37](https://user-images.githubusercontent.com/47163914/52171695-b2684680-2761-11e9-9089-eb272c00260a.png)

**Comment by Messy71** [comment]
[F07ThirdTry-Turn Headland-60m.txt](https://gist.github.com/c51df4408970c51f43dbfc6bbdb477cd.git)
![fsscreen_2019_02_02_18_39_16](https://user-images.githubusercontent.com/47163914/52171759-bd6fa680-2762-11e9-881b-0227abc688be.png)

**Comment by Messy71** [comment]
[F07-FourthTry -TurnHeadland-LEEET m](https://gist.github.com/58efb6637079a5a6ac69f11a4cd605a7.git)
![fsscreen_2019_02_02_18_53_02](https://user-images.githubusercontent.com/47163914/52171835-007e4980-2764-11e9-8ad6-e6c59cbd0f13.png)
![fsscreen_2019_02_02_18_53_23](https://user-images.githubusercontent.com/47163914/52171853-3b807d00-2764-11e9-9ffa-0346646a83d4.png)
![fsscreen_2019_02_02_18_53_33](https://user-images.githubusercontent.com/47163914/52171859-59e67880-2764-11e9-8f27-31998ae77737.png)
![fsscreen_2019_02_02_18_56_01](https://user-images.githubusercontent.com/47163914/52171862-6b2f8500-2764-11e9-83d1-9f67a732cc26.png)
![fsscreen_2019_02_02_18_56_23](https://user-images.githubusercontent.com/47163914/52171868-92865200-2764-11e9-9864-d725a32ccf1d.png)

**Comment by bgo1973** [comment]
Maybe implementing an omega turn using the in advance made headland would make sense. That's how it's done in rl. Reversing is not needed then.

**Comment by Messy71** [comment]
I don’t know how useful it would be because almost all fields have some objects near the headland. If the omega turn will be on the headland course and a house right outside of the corner it will need to be a real flatenned omega. 
I think the Turn mode in the new course generator is the best for work coverage but it needs a lil bit of tweaking.

**Comment by bgo1973** [comment]
The headland must be wide enough for the manouver of course.

**Comment by Messy71** [comment]
fixed for me with v6°01°00073 (i think)
Tested today with v6°01°00076 and i managed to cultivate F07-Ravenport with Fendt 1165MT and Horsch Cruiser 12XL with a 10m width to a proportion of 99.99% (two tiny corners on the curved headland).
I want to mention that i had a minor hiccup, at the transition from the straight lines to first headland CP raised the cultivator, drove to the first waypoint of the headland passes( selected 2 ) and stopped. I just stopped the driver , selected drive from current waypoint , hit drive and he was able to finish the course without other problem. I will do some testing for that and make a new issue / post in an existing one but for now this issue is resolved.
Thank you @pvaiko .
